### PR TITLE
refactor: use the Failover Provider

### PR DESCRIPTION
**Pull request by jtourkos**
(partially) resolves: https://github.com/drips-network/app/issues/1181

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -3,16 +3,31 @@ PORT=string # Optional. The API server port. Defaults to 8080.
...
+
 RPC_URL_POLYGON_AMOY=string # The RPC URL for the Polygon Amoy provider.
 RPC_ACCESS_TOKEN_POLYGON_AMOY=string # Optional. The access token to use for the RPC. If specified, it will be used as a bearer token in the `Authorization` header.
+FALLBACK_RPC_URL_POLYGON_AMOY=string # Optional. The fallback RPC URL for the Polygon Amoy provider. If specified, it will be used if the main RPC URL fails.
+FALLBACK_RPC_ACCESS_TOKEN_POLYGON_AMOY=string # Optional. The access token to use for the fallback RPC. If specified, it will be used as a bearer token in the `Authorization` header.
+
 RPC_URL_FILECOIN=string # The RPC URL for the Filecoin provider.
 RPC_ACCESS_TOKEN_FILECOIN=string # Optional. The access token to use for the RPC. If specified, it will be used as a bearer token in the `Authorization` header.
+FALLBACK_RPC_URL_FILECOIN=string # Optional. The fallback RPC URL for the Filecoin provider. If specified, it will be used if the main RPC URL fails.
+FALLBACK_RPC_ACCESS_TOKEN_FILECOIN=string # Optional. The access token to use for the fallback RPC. If specified, it will be used as a bearer token in the `Authorization` header.
```
Comment (efstajas) [inline .env.template:30]: so many env vars! what do you think about taking a single RPC_CONFIG env var that takes a JSON string with a strict schema that we validate with e.g. `zod` in `appSettings`?

either way, for now I think this is still OK — gonna get a bit crazy when we add more networks soon though.

**Review by efstajas — APPROVED** [review]
LGTM. 👍 Again, maybe think about copying the changes from app's FallbackProvider into here as well.

File: .env.template (hunk 1 above)
Comment (jtourkos) [reply .env.template]: Yes, I agree. Take a look at the latest commit. Was this what you had in mind?

**Review by efstajas — APPROVED** [review]
Before merging make sure to set the env var (or introduce check for env vars in build step)

**Pull request merged**